crates/ruff_linter/src/rules/flake8_simplify/rules/if_else_block_instead_of_dict_lookup.rs

```diff
@@ -66,12 +66,12 @@ pub(crate) fn manual_dict_lookup(checker: &mut Checker, stmt_if: &ast::StmtIf) {
     if ops != &[CmpOp::Eq] {
         return;
     }
-    let [Expr::Constant(ast::ExprConstant {
-        value: constant, ..
-    })] = comparators.as_slice()
-    else {
+    let [expr] = comparators.as_slice() else {
         return;
     };
+    if !expr.is_literal_expr() {
+        return;
+    }
```

Comment:
Here's where a virtual enum could be useful for literal where we could cast the `expr`  to a literal expression (`expr.as_literal_expr`) which would return `Option<T>`.